tests/expr_and_series/rolling_var_test.py

```diff
@@ -0,0 +1,123 @@
+    },
+    "x2": {
+        "kwargs": {"window_size": 3, "min_periods": 1},
+        "expected": [float("nan"), 0.5, 1 / 3, 1.0, 4 / 3, 7 / 3, 3],
+    },
+    "x3": {
+        "kwargs": {"window_size": 2, "min_periods": 1},
+        "expected": [float("nan"), 0.5, 0.5, 2.0, 2.0, 4.5, 4.5],
+    },
+    "x4": {
+        "kwargs": {"window_size": 5, "min_periods": 1, "center": True},
+        "expected": [1 / 3, 11 / 12, 4 / 5, 17 / 10, 2.0, 2.25, 3],
+    },
+    "x5": {
+        "kwargs": {"window_size": 4, "min_periods": 1, "center": True},
+        "expected": [0.5, 1 / 3, 11 / 12, 11 / 12, 2.25, 2.25, 3],
+    },
+    "x6": {
+        "kwargs": {"window_size": 3, "ddof": 2},
+        "expected": [float("nan"), float("nan"), 2 / 3, 2.0, 8 / 3, 14 / 3, 6.0],
+    },
+}
+
+
+@pytest.mark.filterwarnings(
+    "ignore:`Expr.rolling_var` is being called from the stable API although considered an unstable feature."
+)
+def test_rolling_var_expr(
+    request: pytest.FixtureRequest, constructor: Constructor
+) -> None:
+    if "dask" in str(constructor):
+        # TODO(FBruzzesi): Dask is raising the following error:
+        # NotImplementedError: Partition size is less than overlapping window size.
+        # Try using ``df.repartition`` to increase the partition size.
+        request.applymarker(pytest.mark.xfail)
+
+    df = nw.from_native(constructor(data))
+    result = df.select(
+        **{
+            name: nw.col("a").rolling_var(**values["kwargs"])  # type: ignore[arg-type]
+            for name, values in kwargs_and_expected.items()
+        }
+    )
+    expected = {name: values["expected"] for name, values in kwargs_and_expected.items()}
+
+    assert_equal_data(result, expected)
+
+
+@pytest.mark.filterwarnings(
+    "ignore:`Series.rolling_var` is being called from the stable API although considered an unstable feature."
+)
+def test_rolling_var_series(constructor_eager: ConstructorEager) -> None:
+    df = nw.from_native(constructor_eager(data), eager_only=True)
+
+    result = df.select(
+        **{
+            name: df["a"].rolling_var(**values["kwargs"])  # type: ignore[arg-type]
+            for name, values in kwargs_and_expected.items()
+        }
+    )
+    expected = {name: values["expected"] for name, values in kwargs_and_expected.items()}
+    assert_equal_data(result, expected)
+
+
+@given(  # type: ignore[misc]
+    center=st.booleans(),
+    values=st.lists(st.floats(-10, 10), min_size=5, max_size=10),
+)
+@pytest.mark.skipif(PANDAS_VERSION < (1,), reason="too old for pyarrow")
+@pytest.mark.filterwarnings("ignore:.*:narwhals.exceptions.NarwhalsUnstableWarning")
+def test_rolling_var_hypothesis(center: bool, values: list[float]) -> None:  # noqa: FBT001
+    s = pd.Series(values)
+    window_size = random.randint(2, len(s))  # noqa: S311
+    min_periods = random.randint(2, window_size)  # noqa: S311
+    ddof = random.randint(0, min_periods - 1)  # noqa: S311
```

Comment:
Ok, it took me a bit, but I figured out that the different behavior appear when `ddof > min_periods` because the denominator can be negative or zero.

Pandas seems to return NaN indistinctly for windows that do not have enough non null elements and these cases, while polars returns inf (and same in pyarrow since we can control that)